udp/client/clientconn.go

```diff
@@ -682,9 +682,12 @@ func (cc *ClientConn) Process(datagram []byte) error {
 			return
 		}
 
-		// send message with confirmation
+		// send piggybacked response
 		w.response.SetType(udpMessage.Confirmable)
-		w.response.SetMessageID(cc.getMID())
+		if reqType == udpMessage.Confirmable {
+			w.response.SetType(udpMessage.Acknowledgement)
+		}
+		w.response.SetMessageID(reqMid)
```

Comment:
when you `reqType !=  udpMessage.Confirmable` you need to call `w.response.SetMessageID(cc.getMID())` instead of w.response.SetMessageID(cc.getMID()) . otherwise you use same messageId as the peer send and peer can drop the message when he received the response (deduplication).